pkg/vacuum/vacuum.go

```diff
@@ -196,70 +219,148 @@ func (e *Engine) Run(ctx context.Context) {
 			return
 		}
 		log.Info("msg", "compressed chunks need to be vacuumed", "count", len(chunks))
-		p := e.parallelism
-		if len(chunks) < p {
-			// don't spin up more workers than we could possibly use
-			// if parallelism is 6, but we only have 5 chunks to work on, use 5 workers
-			p = len(chunks)
+		numWorkers, err := e.calcNumWorkers(ctx, con, chunks)
+		if err != nil {
+			log.Error("msg", "failed to calc num workers", "error", err)
+			return
 		}
-		runWorkers(ctx, p, chunks, e.worker)
+		runWorkers(ctx, numWorkers, chunks, e.worker)
```

Comment:
Could we sort chunks by chunk age so we first process the oldest? (maybe just ORDER BY in the list chunks query?)

Reply:
the underlying view already sorts oldest to newest. I do think it's probably worth a follow-up PR to move some of the filter logic into the underlying view so that it's not split between connector and extension